R/bayesianProcess.R

```diff
@@ -0,0 +1,619 @@
+    # Add undirected graph for variances and covariances
+    .procResCovGraph(jaspResults, options)
+    # Create lavaan syntax for each model from graph
+    .procModelSyntax(jaspResults, options)
+    # Fit lavaan models based on syntax and dataset and update models container
+    modelsContainer <- .procBayesComputeResults(jaspResults, dataset, options)
+  }
+  # Create container for path plots for each model
+  pathPlotContainer <- .procContainerPathPlots(jaspResults, options)
+  # Create path plots for each model and add to container
+  .procPathPlots(pathPlotContainer, options, modelsContainer)
+  # Create table with model fit indices (AIC, ...)
+  .procBayesModelSummaryTable(jaspResults, options, modelsContainer)
+  # Create container for parameter estimates for each model
+  parEstContainer <- .procContainerParameterEstimates(jaspResults, options, modelsContainer)
+  # Create tables for parameter estimates
+  .procBayesParameterEstimateTables(parEstContainer, options, modelsContainer)
+
+  .procBayesMcmcPlots(jaspResults, options, modelsContainer)
+
+  # Create html output with lavaan syntax for each model
+  .procPlotSyntax(jaspResults, options, modelsContainer)
+
+  return()
+}
+
+.procBayesComputeResults <- function(jaspResults, dataset, options) {
+  modelsContainer <- jaspResults[["modelsContainer"]]
+  nModels <- length(options[["processModels"]])
+
+  for (i in 1:nModels) {
+    modelOptions <- options[["processModels"]][[i]]
+    modelName <- modelOptions[["name"]]
+
+    if (is.null(modelsContainer[[modelName]][["fittedModel"]])) {
+      if (modelOptions[["inputType"]] == "inputModelNumber" && !modelOptions[["modelNumber"]] %in% .procHardCodedModelNumbers()) {
+        fittedModel <- gettextf("%1$s: Hayes model %2$s not implemented", modelName, modelOptions[["modelNumber"]])
+      } else {
+        fittedModel <- .procBayesResultsFitModel(
+          modelsContainer[[modelName]],
+          dataset,
+          options,
+          modelOptions
+        )
+      }
+      state <- createJaspState(object = fittedModel)
+      modelsContainer[[modelName]][["fittedModel"]] <- state
+    }
+  }
+
+  return(modelsContainer)
+}
+
+.procBayesGetPriors <- function(options) {
+  nuPrior   <- sprintf("normal(%s,%s)", options$nuPriorMu, options$nuPriorSigma)
+  betaPrior <- sprintf("normal(%s,%s)", options$betaPriorMu, options$betaPriorSigma)
+  psiPrior  <- sprintf("gamma(%s,%s)[sd]", options$psiPriorAlpha, options$psiPriorBeta)
+  rhoPrior  <- sprintf("beta(%s,%s)[sd]", options$rhoPriorAlpha, options$rhoPriorBeta)
+  return(blavaan::dpriors(nu = nuPrior, beta = betaPrior, psi = psiPrior, rho = rhoPrior))
+}
+
+.procBayesResultsFitModel <- function(container, dataset, options, modelOptions) {
+  # Should model be fitted?
+  doFit <- .procCheckFitModel(container[["graph"]]$object)
+
+  if (!doFit) {
+    dataset <- NULL
+  }
+
+  # Necessary for JASP to find function blavaan
+  blavaan <- blavaan::blavaan
```

Comment:
You explained the reason for doing this. Is this a bug on the jasp side or on the current version of blavaan?

Reply:
It seems to be a bug in `blavaan` specific to JASP. When you run `blavaan` in R, it does not occurr.
The reason is that the `blavaan::bsem` function calls `blavaan` (not `blavaan::blavaan`) under the hood but JASP cannot find the function because it has not loaded the package (as you would do in a normal R session).